### PR TITLE
DTSAM-101 Retire JRD V1 eLinks Logic

### PR DESCRIPTION
**Pull request by mikebrownccd**
https://tools.hmcts.net/jira/browse/DTSAM-101
Retire JRD V1 eLinks Logic

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] Does this PR introduce a breaking change

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java [hunk 1]
```diff
@@ -16,14 +16,10 @@
 import org.springframework.test.web.servlet.MockMvc;
 import org.springframework.test.web.servlet.MvcResult;
 import org.springframework.test.web.servlet.setup.MockMvcBuilders;
-import org.springframework.util.LinkedMultiValueMap;
```
Comment (mattnayler) [inline src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java:19]: This whole file can be removed as all tests in `RefreshControllerIntegrationTest` are replicated in `RefreshControllerIntegrationV2Test` for V2.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java [hunk 2]
```diff
@@ -56,8 +54,8 @@ class RetrieveDataServiceTest {
     private final ParseRequestService parseRequestService = Mockito.mock(ParseRequestService.class);
 
 
-    RetrieveDataService sutJrdV1 = new RetrieveDataService(parseRequestService, crdService, jrdService, false, false);
-    RetrieveDataService sutJrdV2 = new RetrieveDataService(parseRequestService, crdService, jrdService, true, true);
+    RetrieveDataService sutJrdV1 = new RetrieveDataService(parseRequestService, crdService, jrdService);
+    RetrieveDataService sutJrdV2 = new RetrieveDataService(parseRequestService, crdService, jrdService);
```
[the remark this answers is not in the document]
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java]: Done

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java [hunk 3]
```diff
@@ -252,16 +192,14 @@ void shouldReturnJudicialProfileV2() {
...
 
         // GIVEN
         // NB: use local SUT instance, so we can override filter flag.
         var sut = new RetrieveDataService(
             parseRequestService,
             crdService,
-            jrdService,
-            true,
-            v2FilterAuthorisationsByAppointmentId
+            jrdService
```
[the remark this answers is not in the document]
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java]: Done

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java [hunk 4]
```diff
@@ -252,16 +192,14 @@ void shouldReturnJudicialProfileV2() {
 
     @ParameterizedTest()
     @ValueSource(booleans = {true, false})
-    void shouldReturnJudicialProfileV2_withAppointmentsFlag(boolean v2FilterAuthorisationsByAppointmentId) {
+    void shouldReturnJudicialProfileV2_withAppointmentsFlag() {
```
[the remark this answers is not in the document]
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java]: Done

File: src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java (hunk 1 above)
Comment (mikebrownccd) [reply src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java:19]: RefreshControllerIntegrationV2Test i think has been renamed to RefreshControllerRefreshJobIntegrationTest as part of review comments

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java]: Done

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java [hunk 5]
```diff
@@ -278,29 +216,11 @@ void shouldReturnJudicialProfileV2_withAppointmentsFlag(boolean v2FilterAuthoris
             assertNotNull(k);
             assertNotNull(v);
             v.forEach(userAccessProfile -> {
-                if (v2FilterAuthorisationsByAppointmentId) {
-                    // FILTERED: only 2 authorisations from "judicialProfileSampleV2.json" attached to each appointment
```
[the remark this answers is not in the document]
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java:282]: Done

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java [hunk 6]
```diff
@@ -513,27 +487,6 @@ public static JudicialProfileV2 buildJudicialProfileWithParamsV2(
                 .build();
     }
 
-    public static Appointment buildAppointmentWithParams(String epimms, String isPrinciple, String appointment,
```
[the remark this answers is not in the document]
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java:516]: Done

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/AssignmentRequestBuilderTest.java [hunk 7]
```diff
@@ -524,7 +476,7 @@ void convertUserProfileToJudicialAccessProfileV2_withDiffTicketCode(boolean filt
 
             // WHEN
             Set<UserAccessProfile> judicialAccessProfiles = AssignmentRequestBuilder
-                    .convertProfileToJudicialAccessProfileV2(judicialProfile, filterAuthorisationsByAppId);
+                    .convertProfileToJudicialAccessProfileV2(judicialProfile);
```
[the remark this answers is not in the document]
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/AssignmentRequestBuilderTest.java:478]: Done

File: src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java (hunk 1 above)
Comment (mikebrownccd) [reply src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java:19]: but the files look to contain different tests, please take another tlook

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/AssignmentRequestBuilderTest.java [hunk 8]
```diff
@@ -557,7 +509,7 @@ void convertUserProfileToJudicialAccessProfileV2_withInactiveTicketCode(boolean
...
```
[the remark this answers is not in the document]
Comment (mikebrownccd) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/AssignmentRequestBuilderTest.java:510]: Done

File: src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java (hunk 1 above)
Comment (mikebrownccd) [reply src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java:19]: After discussing with Matt I agree that the test class can be removed, and it has been. Done

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java]: Resolved in f66b5f327965c723850f68c07995636010b47c09.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java]: Resolved in f66b5f327965c723850f68c07995636010b47c09.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RetrieveDataServiceTest.java]: Resolved in f66b5f327965c723850f68c07995636010b47c09.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/TestDataBuilder.java:516]: Resolved in f66b5f327965c723850f68c07995636010b47c09.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/AssignmentRequestBuilderTest.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/AssignmentRequestBuilderTest.java:476]: Resolved in 08b92e18bd8f8e5f7d77e6e11a0d0e0d301f064b.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/AssignmentRequestBuilderTest.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/helper/AssignmentRequestBuilderTest.java:508]: Resolved in f66b5f327965c723850f68c07995636010b47c09.

**Review by mattnayler — APPROVED** [review]
LGTM

**Review by abul-syed — APPROVED** [review]
lgtm

**Pull request merged**